scan-triggers/aws-typescript-bucket-full-scan/cloudone-filestorage-plugin-full-scan.json

```diff
@@ -0,0 +1,240 @@
er queue region: ' + sqs_region)\nlogger.info('Scanner queue endpoint URL: ' + sqs_endpoint_url)\nlogger.info('SNS Arn: ' + os.environ['SNSArn'])\n\n  \ndef get_matching_s3_objects(bucket, prefix=\"\", suffix=\"\"):\n  s3 = boto3.client(\"s3\")\n  paginator = s3.get_paginator(\"list_objects_v2\")\n    \n  kwargs = {'Bucket': bucket}\n    \n  # We can pass the prefix directly to the S3 API. If the user has passed\n  # a tuple or list of prefixes, we go through them one by one.\n  if isinstance(prefix, str):\n    prefixes = (prefix,)\n  else:\n    prefixes = prefix\n    \n  for key_prefix in prefixes:\n    kwargs[\"Prefix\"] = key_prefix\n    \n  for page in paginator.paginate(**kwargs):\n    try:\n      contents = page[\"Contents\"]\n    except KeyError:\n      break\n    \n  for obj in contents:\n    key = obj[\"Key\"]\n    if key.endswith(suffix):\n      yield obj\n  \n  \n  \ndef create_presigned_url(bucket_name, object_name, expiration):\n  \"\"\"Generate a presigned URL to share an S3 object\n    \n  :param bucket_name: string\n  :param object_name: string\n  :param expiration: Time in seconds for the presigned URL to remain valid\n  :return: Presigned URL as string. If error, returns None.\n  \"\"\"\n    \n  # Generate a presigned URL for the S3 object\n  s3_client = boto3.client('s3', config=Config(s3={'addressing_style': 'virtual'}, signature_version='s3v4'))\n  try:\n    response = s3_client.generate_presigned_url(\n      'get_object',\n      Params={\n      'Bucket': bucket_name,\n      'Key': object_name\n    },\n    ExpiresIn=expiration\n    )\n  except ClientError as e:\n    print('failed to generate pre-signed URL: ' + str(e))\n    return None\n    \n  # The response contains the presigned URL which is sensitive data\n  return response\n  \n\ndef push_to_sqs(bucket_name, object_name, presigned_url, event_time):\n  object = {\n    'S3': {\n    'bucket': \"{0}\".format(bucket_name),\n    'object': \"{0}\".format(object_name)\n  },\n    'ScanID': str(uuid.uuid4()),\n    'SNS' : os.environ['SNSArn'],\n    'URL': \"{0}\".format(presigned_url),\n    'ModTime': event_time\n  }\n  try:\n    session = boto3.session.Session(region_name=sqs_region)\n    sqs = session.resource(service_name='sqs', endpoint_url=sqs_endpoint_url)\n    queue = sqs.Queue(url=sqs_url)\n    response = queue.send_message(MessageBody=json.dumps(object))\n    return response\n  except ClientError as e:\n    print('failed to push SQS message: ' + str(e))\n    return None\n  \ndef handler(event, context):  \n  for object in get_matching_s3_objects(bucket=os.environ['BucketToScanName']):\n    if object['Size'] == 0:\n      continue\n    presigned = create_presigned_url(bucket_name=bucket, object_name=object[\"Key\"], expiration=3600)\n    sqs_response = push_to_sqs(bucket, object[\"Key\"], presigned, object['LastModified'].isoformat())\n    print(sqs_response)\n  return {\n    'statusCode': 200,\n    'body': json.dumps('Manual Scan was successfully triggered.')\n  }\n      "
```

Comment:
This is python. But the folder remains typescript.

Reply:
In the function `get_matching_s3_objects`, the second loop `for obj in contents:` should be inside the first loop  `for page in paginator.paginate(**kwargs):` in order to iterate over the pages, otherwise, only the last page is iterated.

```python
def get_matching_s3_objects(bucket, prefix="", suffix=""):
  # ...
  for page in paginator.paginate(**kwargs):
    try:
      contents = page["Contents"]
    except KeyError:
      break

    for obj in contents:
      key = obj["Key"]
      if key.endswith(suffix):
        yield obj
```

Reply:
Hi @raphabot, we are publishing all plugins with our pipeline in FSS AWS account with author named Trend Micro Cloud One File Storage Security. It would be more unified for customers and we are trying to get a verified author badge. Also if this plugin is published by you, you will have to monitor this plugin and publish it when a change is merged, which I think would be a burden on you.

Reply:
@raphabot,
Bring your publication offline is not a must. What we need is a metadata section in the template like this

https://github.com/trendmicro/cloudone-filestorage-plugins/blob/257eb76144279c1a85777353c3d7b012ded44e3c/post-scan-actions/aws-python-promote-or-quarantine/template.yml#L4-L17